scripts/github-repos/github-repos.rb

```diff
@@ -7,220 +7,408 @@
+
+    opt.separator ""
+    opt.separator "Commands:"
+    opt.separator "    invite: Create a team called STUDENTTEAM under the org..."
+    opt.separator "            If STUDENTTEAM is already exist, the script will resent invitation to students."
+    opt.separator "            (Temporary for special situations in Fa23)"
+    opt.separator "            If PREFIX is provided, it will assume the csv file contains \"Team\" column, and create child teams under the STUDENTTEAM, and invites them to child teams.\n"
+    opt.separator "    create_teams: Assuming students are in STUDENTTEAM, create child teams for students for CHIP 10.5 and add them to the child team. \n"
+    opt.separator "    indiv_repos: Create CHIPS repo for each stedent in STUDENTTEAM. Repos' names are form like \"PREFIX-[username]-FILENAME\"\n"
+    opt.separator "    team_repos: Create 10.5 repos for each child team. Repos' names are form like \"PREFIX-FILENAME-[Team number]\""
+    opt.separator "                Make sure child teams are formed before running this command.\n"
+    opt.separator "    remove_indiv_repos: Delete all repos whose names are formed like \"PREFIX-[username]-FILENAME\".\n"
+    opt.separator "    remove_team_repos: Delete all repos whose names are formed like \"PREFIX-FILENAME-[Team number]\".\n"
+    opt.separator "    remove_teams: Remove all students and child teams in STUDENTTEAM from the org. Remove STUDENTTEAM as well.\n"
+    opt.separator "    remove_team_repo_access: Remove students access to CHIP 10.5 repos that are formed like \"PREFIX-FILENAME-[Team number]\".\n"
+    opt.separator "Options:"
+
+    opt.on('-cCSVFILE', '--csv=CSVFILE', 'CSV file containing at leaset "Username" named columns') do |csv|
+      org.csv = csv
+    end
+    opt.on('-oORGNAME', '--orgname=ORGNAME', 'The name of the org') do |orgname|
+      org.orgname = orgname
+    end
+    opt.on('-sSTUDENTTEAM', '--studentteam=STUDENTTEAM', 'The team name of all the students team') do |studentteam|
+      org.parentteam = studentteam
+    end
+    opt.on('-pPREFIX', '--prefix=PREFIX', 'Semester prefix, eg fa23.') do |pfx|
+      org.semester = pfx
     end
-    $opts.parse!
-    command = ARGV.pop
-    case command
-    when 'invite' then org.invite
-    when 'repos' then org.create_repos 
-    when 'remove' then org.remove
-    when 'remove_access' then org.remove_access
-    else org.print_error
-    end
-    puts "Run successfully."
-    puts "Script ends."
+    opt.on('-fFILENAME', '--filename=FILENAME', 'The base filename for repos') do |filename|
+      org.base_filename = filename
+    end
+    opt.on('-tTEMPLATE', '--template=TEMPLATE', 'The repo name within the org to use as template') do |template|
+      org.template = template
+    end
+    opt.on('-gGSITEAM', '--gsiteam=GSITEAM', 'The team name of all the staff team') do |gsiteam|
+      org.gsiteam = gsiteam
+    end
+  end
+  $opts.parse!
+  command = ARGV.pop
+  case command
+  when 'invite' then org.invite

```

Comment:
i'd suggest rewriting this code along the following lines.  define a class method in `OrgManager` that returns a valid list of subcommand names (invite, create_teams, etc) as a list of strings or symbols.  then this code can just say:
```ruby
if OrgManager.commands.include?(command)
  org.send(command)
else
  org.print_error
end
```